### PR TITLE
ref(py3): Use __str__ over __unicode__ in auth/provider

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

**Review by dcramer — COMMENTED** [review]
wont this cause issues in some situations? i thought unicode was safe with py3 still (or theres some wrapper in six)

File: src/sentry/auth/provider.py [hunk 1]
```diff
@@ -15,9 +19,8 @@ class MigratingIdentityId(namedtuple("MigratingIdentityId", ["id", "legacy_id"])
 
     __slots__ = ()
 
-    def __unicode__(self):
-        # Default to id when coercing for query lookup
-        return self.id
+    def __str__(self):
+        return force_text(self.id)
```
Comment (mattrobenolt) [inline src/sentry/auth/provider.py:23]: Isn’t force_text spitting back unicode? So wouldn’t this be wrong for py2?

Comment (EvanPurkhiser) [reply src/sentry/auth/provider.py:23]: Yeah, that's what `python_2_unicode_compatible` fixes.

https://consideratecode.com/2017/08/25/what-does-python_2_unicode_compatible-do/

Comment (mattrobenolt) [reply src/sentry/auth/provider.py:23]: Welp.

> When using the decorator, __str__ should always return a unicode string. Therefor we cast the string to text_type.

That’s unintuitive.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**